### PR TITLE
Im a bot option in onboarding

### PR DESCRIPTION
**Pull request by TheYande**
when a user selects this option in onboarding, they are automaticly kicked

this is ment to automaticly kick bots, becuase they usually pick every possible option

### CODE REVIEW

File: modules/joins.ts [hunk 1]
```diff
@@ -91,6 +91,14 @@ defineEvent("guildMemberAdd", async (member) => {
 		.catch(() => void 0);
 });
 
+defineEvent("guildMemberUpdate", async (_, member) => {
+	if (!config.roles.autoKick) return;
+	if (member.roles.resolve(config.roles.autoKick.id))
```
Comment (RedGuy12) [inline modules/joins.ts:96]:
```suggestion
	if (!member.roles.resolve(config.roles.autoKick.id)) return

File: modules/joins.ts [hunk 2]
```diff
@@ -91,6 +91,14 @@ defineEvent("guildMemberAdd", async (member) => {
...
+		if (!member.joinedTimestamp || member.joinedTimestamp > Date.now() - 60 * 60 * 1000)
```
Comment (RedGuy12) [inline modules/joins.ts:97]: u were right
```suggestion
		if (member.joinedTimestamp && member.joinedTimestamp > Date.now() - 60 * 60 * 1000)

File: modules/joins.ts [hunk 3]
```diff
@@ -91,6 +91,14 @@ defineEvent("guildMemberAdd", async (member) => {
...
+			member.kick("chose the autokick role in onboarding");
+		else member.roles.remove(config.roles.autoKick, "chose autokick option after 1 hour");
```
Comment (RedGuy12) [inline modules/joins.ts:99]:
```suggestion
			member.kick("Autokick role assigned");
		else member.roles.remove(config.roles.autoKick, "Autokick role assigned post-join");

File: modules/joins.ts (hunk 2 above)
Comment (TheYande) [reply modules/joins.ts:97]: 💀

**Comment by RedGuy12** [comment]
lint is failing bc you need to `await`

**Comment by TheYande** [comment]
link can suck my ... ok sure

**Comment by RedGuy12** [comment]
it's import to always `await` - otherwise if it errors for whatever reason the error log will just say "hey uh this error occurred somewhere" instead of saying "...in the autoKick code at this line number". depending on how the code is set up, it could even crash the entire bot

**Comment by TheYande** [comment]
can you do the suggestinog thingy im lazy

File: modules/joins.ts [hunk 4]
```diff
@@ -91,6 +91,15 @@ defineEvent("guildMemberAdd", async (member) => {
 		.catch(() => void 0);
 });
 
+defineEvent("guildMemberUpdate", async (_, member) => {
+	if (!config.roles.autoKick) return;
+	if (!member.roles.resolve(config.roles.autoKick.id)) return;
+
+	if (member.joinedTimestamp && member.joinedTimestamp > Date.now() - 60 * 60 * 1000)
+		member.kick("Autokick role assigned");
```
Comment (TheYande) [inline modules/joins.ts:99]:
```suggestion
await member.kick("Autokick role assigned");
```

**Comment by TheYande** [comment]
ok i un locked out myself

File: modules/joins.ts [hunk 5]
```diff
@@ -91,6 +91,15 @@ defineEvent("guildMemberAdd", async (member) => {
 		.catch(() => void 0);
 });
 
+defineEvent("guildMemberUpdate", async (_, member) => {
+	if (!config.roles.autoKick) return;
+	if (!member.roles.resolve(config.roles.autoKick.id)) return;
+
+	if (member.joinedTimestamp && member.joinedTimestamp > Date.now() - 60 * 60 * 1000)
+		await member.kick("Autokick role assigned");
+	else member.roles.remove(config.roles.autoKick, "Autokick role assigned post-join");
```
Comment (TheYande) [inline modules/joins.ts:100]:
```suggestion
else await member.roles.remove(config.roles.autoKick, "Autokick role assigned post-join");
```

**Comment by RedGuy12** [comment]
@TheYande Your code has a syntax error.
After you fix that, please fix the remaining lint error. This can be done automatically with `npm run lint -- --fix`. Finally, please gate this feature behind a flag - see #524 and 1a3c8891b9685b8d8ca43dd7b2da4ddb400be0d5.

**Pull request merged**